### PR TITLE
Bug fix for coral MessageService wrapper

### PR DESCRIPTION
**Pull request by ggovi**
#### PR description:

This pr addresses https://github.com/cms-sw/cmssw/issues/31896

The fixes involve a wrapper for the Coral message Reporter:
- complying with expected pointer ownership in coral
-  ensuring better hand-shake while de-scoping objects

#### PR validation:

CondDB Unit test, integration test with BeamSpot client ( including the recipe for reproducing the above issue )

### CODE REVIEW

**Comment by jfernan2** [comment]
Dear @ggovi 
Could we have a backport of this PR to 11_2_X in order to use it at P5 during next MWGR?
Thanks

**Comment by ggovi** [comment]
please test

File: CondCore/CondDB/src/ConnectionPool.cc [hunk 1]
```diff
@@ -23,6 +23,9 @@ namespace cond {
 
     ConnectionPool::ConnectionPool() {
       m_pluginManager = new cond::CoralServiceManager;
+      m_msgReporter = new CoralMsgReporter;
+      // ownership is acquired ( instrinsically unsafe, need to change coral )
```
Comment (christopheralanwest) [inline CondCore/CondDB/src/ConnectionPool.cc:27]: Could you correct the spelling of "intrinsically"?

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
please test

**Comment by ggovi** [comment]
+1

**Comment by christopheralanwest** [comment]
+1

**Comment by silviodonato** [comment]
+1
@cms-sw/core-l2 the ASAN IB should look much better after this PR

**Pull request merged**